### ISSUE TITLE
Many-to-many light-opsin support

### ISSUE THREAD

**Issue by kjohnsen** [Sensory-Information-Processing-Lab/Cleo#3]
Currently, only one OptogeneticIntervention could be used on a neuron group at once. This is because only one Iopto gets added. We could support more by making Iopto a sum of currents identified by names, e.g., Iopto_name.

**Comment by kjohnsen** [comment]
Yes, an epsilon dictionary parameter would define sensitivity to different wavelengths. For ChR2, for example, `epsilon = {473: 1, 620: 0.5}` or something like that, meaning it responds half as much to 620-nm light as to 473-nm light.

**Comment by kjohnsen** [comment]
My best idea for implementing this refactor so far: make an opsin an InterfaceDevice and use an `inject` function syntax for both opsin and light, e.g.:
```python
sim.inject(opsin, ng1, ng2, **kwargs)
sim.inject(light1, ng1, ng2)
sim.inject(light2, ng1, ng2)
```

An Opsin, when connected to an NG, would set up a light aggregator neuron for each target neuron and a Synapses implementing the opsin model as before, but from the light neurons to the target neurons, instead of target->target. The target neuron would keep track of transformed opsins with a dynamic attribute for the next step:

Injecting a LightDevice to a neuron would set up a Synapses object from each light source in the LightDevice to the light aggregator neurons of each opsin in the target neuron group. E.g., for an array of 10 LEDs, there would be 10 Synapses objects, one each onto the ChR2 light aggregator NG of a given target NG. If the target NG had a second opsin, e.g., Jaws, there would be another 10 Synapses objects, each onto the light aggregator NG of the Jaws opsin for the target NG

**Comment by kjohnsen** [comment]
Could possibly have a current aggregator NG, so Light->light aggregator--(opsin model)—>current aggregator

**Comment by kjohnsen** [comment]
Try attrs for parameter-heavy classes. 

Structure:

Light - for single or multi-source device
- takes LightModel, coordinates, orientation

LightModel
- abstract, functions for T and visualization 

FiberModel, LEDModel, etc. 
- concrete, contains default params 

Don't subclass for specific parameters---instead provide a function like fiber473nm() that constructs a FiberModel with wavelength-dependent params.

**Comment by kjohnsen** [comment]
More thoughts from notes on phone:
- LightModel would have T function, as well as viz point generation function

**Comment by kjohnsen** [comment]
implemented in #33
